Widgets_CDAT/textStyleEditorWidget/TextStyleEditor.py

```diff
@@ -0,0 +1,253 @@
+
+
+        font_size_row = QtGui.QHBoxLayout()
+        font_size_row.addWidget(font_label)
+        font_size_row.addWidget(self.font_box)
+        font_size_row.addWidget(size_label)
+        font_size_row.addWidget(self.size_box)
+        font_size_row.insertStretch(2, 3)
+
+        # Save and Cancel Buttons
+
+        cancel_button = QtGui.QPushButton()
+        cancel_button.setText("Cancel")
+        cancel_button.clicked.connect(lambda: self.close())
+
+        saveas_button = QtGui.QPushButton()
+        saveas_button.setText("Save As")
+        saveas_button.clicked.connect(self.saveAs)
+
+        save_button = QtGui.QPushButton()
+        save_button.setText("Save")
+        save_button.clicked.connect(self.save)
+
+        save_cancel_row = QtGui.QHBoxLayout()
+        save_cancel_row.addWidget(cancel_button)
+        save_cancel_row.addWidget(saveas_button)
+        save_cancel_row.addWidget(save_button)
+        save_cancel_row.insertStretch(1, 1)
+
+        # Set up wrap
+        wrap.addWidget(self.preview)
+        wrap.addLayout(align_angle_row)
+        wrap.addLayout(font_size_row)
+        wrap.addLayout(save_cancel_row)
+        self.setLayout(wrap)
+
+
+    def setTextObject(self, textObject):
+        self.textObject = textObject
+        self.preview.setTextObject(self.textObject)
+        self.setWindowTitle("Edit Style \"" + self.textObject.name + "\"")
+
+        # set initial values
+        cur_valign = self.textObject.valign
+        for button in self.VAGroup.buttons():
+            if cur_valign == 0 and button.text() == "Top":
+                button.setChecked(True)
+            elif cur_valign == 2 and button.text() == "Mid":
+                button.setChecked(True)
+            elif cur_valign == 4 and button.text() == "Bot":
+                button.setChecked(True)
+
+        cur_halign = self.textObject.halign
+        for button in self.HAGroup.buttons():
+            if cur_halign == 0 and button.text() == "Left":
+                button.setChecked(True)
+            elif cur_halign == 1 and button.text() == "Center":
+                button.setChecked(True)
+            elif cur_halign == 2 and button.text() == "Right":
+                button.setChecked(True)
+
+        self.angle_slider.setSliderPosition(self.textObject.angle)
+
+        self.size_box.setValue(self.textObject.height)
+
+
+    def updateButton(self, button):
+        if button.text() == "Top":
+            self.textObject.valign = "top"
+
+        elif button.text() == "Mid":
+            self.textObject.valign = "half"
+
+        elif button.text() == "Bot":
+            self.textObject.valign = "bottom"
+
+        elif button.text() == "Left":
+            self.textObject.halign = "left"
+
+        elif button.text() == "Center":
+            self.textObject.halign = "center"
+
+        elif button.text() == "Right":
+            self.textObject.halign = "right"
+
+        self.preview.setTextObject(self.textObject)
```

Comment:
You should just call 

```python
self.preview.update()
```

unless the preview hasn't yet been assigned the `textObject`. It will automatically grab the relevant features, and be less flicker-y.